ql/src/java/org/apache/hadoop/hive/ql/DriverUtils.java

```diff
@@ -61,6 +61,7 @@ public static void runOnDriver(HiveConf conf, String user,
       throw new IllegalArgumentException(JavaUtils.txnIdToString(compactorTxnId) +
           " is not valid. Context: " + query);
     }
+    sessionState.setSyncCleanup();
```

Comment:
this is a one-way switch; so there is no way to switch back to the earlier behaviour

Reply:
This switch is actually meant to be constant for a specific session. I've put it as a setter as it seemed easier than putting it in constructor(many changes to many file). Sync cleanup is used for compactor and stats updater thread. Everything else will be async. I believe compactor and stats updater use their own session repeatedly. Unless a regular query re-uses compactor/statsupdater's session, there is no need to set it back to async. I'll double check if this is the case.